utils.js

```diff
@@ -4759,11 +4759,87 @@ export async function parseCargoTomlData(cargoTomlFile, simple = false) {
   }
 }
 
-export async function parseCargoData(cargoData) {
+/**
+ * Parse a Cargo.lock file to find components within the Rust project.
+ *
+ * @param {string} cargoLockFile A path to a Cargo.lock file. The Cargo.lock-file path may be used as information for extended attributes, such as manifest based evidence.
+ * @param {boolean} simple Return a simpler representation of the component by skipping extended attributes and license fetch.
+ *
+ * @returns {array} A list of the project's components as described by the Cargo.lock-file.
+ */
+export async function parseCargoData(cargoLockFile, simple = false) {
+  const addPackageToList = (packageList, newPackage, { simple }) => {
+    if (!newPackage) {
+      return;
+    }
+
+    const purl = new PackageURL(
+      "cargo",
+      "",
+      newPackage.name,
+      newPackage.version,
+      null,
+      null
+    ).toString();
+    const component = {
+      type: "library",
+      group: newPackage.group,
+      "bom-ref": purl,
+      purl: purl,
+      name: newPackage.name,
+      version: newPackage.version
+    };
+
+    if (newPackage._integrity) {
+      component.hashes = [
+        {
+          alg: "SHA-384",
+          content: pkg._integrity
+        }
+      ];
+    }
+
+    if (!simple) {
+      // Assign evidence according to CycloneDX's confidence recommendations in section Evidence of:
+      // * https://cyclonedx.org/guides/OWASP_CycloneDX-Authoritative-Guide-to-SBOM-en.pdf
+      // The evidence is deemed to be reliable because Cargo itself generates
+      // the Cargo.lock-file based on the listed dependencies in the
+      // Cargo.toml-file and registry information. So, either we get a direct
+      // dependency (very likely), or a transitive dependency based on
+      // evidence from the package information in the Cargo registry.
+      component.evidence = {
```

Comment:
We also add a property called `SrcFile` for users stuck with 1.4 spec.

Reply:
Turns out the comment chain below sort of blocked the `SrcFile`-object. It's added! From line 4823.